### ISSUE TITLE
Type override for attributes is not checked

### ISSUE THREAD

**Issue by dmoisset** [python/mypy#2211]
from what I see, overriding an attribute with an incompatible type (which can lead to unsoundness) does not produce any kind of warning with mypy:

``` python
class Base:
    attr = "foo"

    def f(self) -> str:
        return self.attr

class Derived(Base):
    attr = 42

d = Derived()

reveal_type(d.f())
```

The code above shows that mypy thinks that the `d.f()` expression is a string, while it's actually an int. Shouldn't this kind of redefinitions be forbidden?

**Comment by gvanrossum** [comment]
IIRC there's an issue for this already. It's definitely a bug.

--Guido (mobile)

**Comment by dmoisset** [comment]
I have looked before opening but couldn't find one... After a second look, there's a mention to this in #1338 , but they seem to be two related but distinct issues.

**Comment by gvanrossum** [comment]
Still I suspect that they can be fixed with one PR. Shouldn't even be that
hard.

**Comment by gvanrossum** [comment]
Fixed by #2510.
